### PR TITLE
feat(doc): generate doc with both versions of pydantic

### PR DESCRIPTION
**Pull request by tcrasset**
- feat(doc): make doc work with both versions of pydantic
- fix(doc): add blank line to allow generation of docstring
- docs(docs): document how we use sphinx to generate documentation

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+The objective of this file is to describe how the build process of the documentation.
+
+## Tools
+
+We use `sphinx` to build our documentation, along with a few extensions.
+
+- `sphinx.ext.autodoc`
+
+  TODO: Explain to which files this applies to.
```
Comment (tcrasset) [inline doc/README.md:13]: Not the scope of this PR, I mainly wanted a place to write down some troubleshooting steps I found.

File: avatars/processors/inter_record_range_difference.py [hunk 2]
```diff
@@ -79,6 +79,7 @@ class InterRecordRangeDifferenceProcessor:
     ...         "value_difference": [0, 2, 0, 4, 0, 5],
     ...     }
     ... )
+
     >>> processor.postprocess(df, preprocessed_df)
```
Comment (tcrasset) [inline avatars/processors/inter_record_range_difference.py:83]: `sphinx` complained about wrong indent

File: doc/source/conf.py [hunk 3]
```diff
@@ -78,8 +78,28 @@
...
+# and, although we can put custom code in `conf.py` that get's executed at
+# import, `sphinx-multiversion` only takes `conf.py` of the version you're building
+# _from_.
+# To circumvent that, we whitelist the tags that were at pydantic1 and have the following
+# workflow:
+# - Uninstall pydantic2 and install pydantic1
+# - Run `sphinx-multiversion` with the default values (the tag whitelist here below)
+# - Uninstall pydantic1 and install pydantic2
+# - Run `sphinx-multiversion` with a configuration override via the CLI
+# to build the documentation
```
Comment (tcrasset) [inline doc/source/conf.py:100]: Not sure if this belongs here, or in `README.md`

File: Makefile [hunk 4]
```diff
@@ -77,10 +77,27 @@ doc-build:  ## Build the docs
...
 	poetry run python doc/bin/modify_class_name.py $(DOC_OUTPUT_DIR)
 .PHONY: doc-build
 
+doc-build-pydantic-v1:
+	poetry remove pydantic autodoc-pydantic
+	poetry add 'pydantic<2' 'autodoc-pydantic<2'
+	git restore -- pyproject.toml poetry.lock # restoring straight after install makes sure that any errors during doc build will not leave them dirty
+	# We whitelist no branches, as they will be built by the pydantic-v1 target
+	poetry run sphinx-multiversion $(DOC_SOURCE_DIR) $(DOC_OUTPUT_DIR) -D 'smv_branch_whitelist=None'
+.PHONY: doc-build-pydantic-v1
```
Comment (tas17) [inline Makefile:91]: This is temporary ?

File: doc/source/conf.py (hunk 3 above)
Comment (tas17) [reply doc/source/conf.py:100]: Yeah, more in the Readme in its own section

**Review by tas17 — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (tcrasset) [reply Makefile:91]: Currently, we regenerate the documentation for every version. We could move to only doing new versions and not touch the old ones. In the current state, this is thus not temporary.

**Pull request merged**